### PR TITLE
sources: refactor base sources into new package.

### PR DESCRIPTION
**Pull request by kyrofa**
This PR makes progress on LP: [#1648906](https://bugs.launchpad.net/snapcraft/+bug/1648906) by moving `snapcraft/internal/sources.py` into the `sources` package, and extracting `Base` and `FileBase` into their own module. This is the first in a series of PRs; subsequent onces will refactor the other sources into their own modules.

### CODE REVIEW

**Comment by kyrofa** [comment]
@elopio can you please re-trigger xenial autopkg tests here?

**Comment by sergiusens** [comment]
done @kyrofa

**Comment by elopio** [comment]
retried.

File: snapcraft/internal/new_sources/__init__.py [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from snapcraft.internal.new_sources._base import (  # noqa
```
Comment (sergiusens) [inline snapcraft/internal/new_sources/__init__.py]: this should be relative,

```python
from ._base import (
```

File: setup.py [hunk 2]
```diff
@@ -42,6 +42,7 @@
               'snapcraft.internal',
               'snapcraft.internal.cache',
               'snapcraft.internal.deltas',
+              'snapcraft.internal.new_sources',
```
Comment (sergiusens) [inline setup.py]: still call this `sources`, I don't see a reason for `new_sources`. The migration could be transparent and if it isn't it doesn't matter *as much* as it lives in `internal`

File: snapcraft/internal/new_sources/__init__.py (hunk 1 above)
Comment (kyrofa) [inline snapcraft/internal/new_sources/__init__.py]: Done.

File: setup.py (hunk 2 above)
Comment (kyrofa) [inline setup.py]: :+1: , I didn't realize GitHub would follow renames without dumping a bunch of red and green. This is a much better path than the intermediate package.

**Pull request closed without merging**